packages/node/src/indexer/indexer.manager.ts

```diff
@@ -169,30 +178,128 @@ export class IndexerManager {
     }
   }
 
-  private getStartBlockFromDataSources() {
-    const startBlocksList = this.getDataSourcesForSpecName().map(
-      (item) => item.startBlock ?? 1,
-    );
-    if (startBlocksList.length === 0) {
-      logger.error(
-        `Failed to find a valid datasource, Please check your endpoint if specName filter is used.`,
+  private async ensureProject(): Promise<string> {
+    let schema = await this.getProjectSchema();
+    if (!schema) {
+      schema = await this.createProjectSchema();
+    } else {
+      if (argv['force-clean']) {
+        try {
+          // drop existing project schema and metadata table
+          this.sequelize.dropSchema(schema, {
+            logging: false,
+            benchmark: false,
+          });
+
+          // remove schema from subquery table (might not exist)
+          await this.sequelize.query(
+            ` DELETE
+              FROM public.subqueries
+              where db_schema = :subquerySchema`,
+            {
+              replacements: { subquerySchema: schema },
+              type: QueryTypes.DELETE,
+            },
+          );
+
+          logger.info('force cleaned schema and tables');
+        } catch (err) {
+          logger.error(err, 'failed to force clean schema and tables');
+        }
+        schema = await this.createProjectSchema();
+      }
+    }
+    return schema;
+  }
+
+  private async getProjectSchema(): Promise<string> {
+    let schema = argv.schema;
+    if (!schema) {
+      schema = this.nodeConfig.subqueryName;
+    }
+    const schemas = (await this.sequelize.showAllSchemas(
+      undefined,
+    )) as unknown as string[];
+    if (!schemas.includes(schema)) {
+      const result = await this.sequelize.query(
+        `select db_schema from public.subqueries where name = :name`,
+        {
+          replacements: { name: schema },
+          type: QueryTypes.SELECT,
+        },
       );
-      process.exit(1);
+      if (result.length === 0) {
+        schema = undefined;
```

Comment:
if can not find this schema in db, we should use argv.schema or project name as its schema name

Reply:
this method is to fetch a schema for the project if it exists, otherwise it will return undefined, if I make this change it will mean that I can no longer detect if a project must be created using `createProjectSchema`